bufr/src/main/java/ucar/nc2/iosp/bufr/BufrIospBuilder.java

```diff
@@ -123,7 +124,11 @@ private void addStructure(Group.Builder group, Structure.Builder parent, BufrCon
     dkey.name = uname; // name may need to be changed for uniqueness
 
```

Comment:
I don't know that this method of catching will work out as well in bufr, especially with structs. I would lean toward letting bufr throw throw the error if the dimension is 0, since the issue only complained about hdf5 datafiles.